examples/gno.land/p/moul/txlink/txlink.gno

```diff
@@ -22,15 +39,99 @@ import (
 
 const chainDomain = "gno.land" // XXX: std.ChainDomain (#2911)
 
+// Realm represents a specific realm for generating tx links.
+type Realm string
+
+// TxBuilder provides a fluent interface for building transaction URLs
+type TxBuilder struct {
+	fn    string   // function name
+	args  []string // key-value pairs
+	send  string   // optional send amount
+	realm Realm    // realm for the URL
+}
+
+// NewLink creates a transaction link builder for the specified function in the current realm.
+func NewLink(fn string) *TxBuilder {
+	return Realm("").NewLink(fn)
+}
+
+// NewLink creates a transaction link builder for the specified function in this realm.
+func (r Realm) NewLink(fn string) *TxBuilder {
+	if fn == "" {
+		return nil
+	}
+	return &TxBuilder{fn: fn, realm: r}
+}
+
+// AddArg adds a key-value argument pair. Returns the builder for chaining.
+func (b *TxBuilder) AddArg(key, value string) *TxBuilder {
+	if b == nil {
+		return nil
+	}
+	if key == "" {
+		return b
+	}
+
+	// Special case: ".send" is a reserved keyword.
+	if key == ".send" {
+		if b.send == "" {
+			return b.AddSend(value)
+		}
+		return b
+	}
+
+	b.args = append(b.args, key, value)
+	return b
+}
+
+// AddArgs adds multiple key-value pairs at once. Arguments should be provided
+// as pairs: AddArgs("key1", "value1", "key2", "value2").
+func (b *TxBuilder) AddArgs(args ...string) *TxBuilder {
+	if b == nil {
+		return nil
+	}
+	if len(args)%2 != 0 {
+		return b.AddArg("error", "odd number of arguments")
+	}
+	// Add key-value pairs
+	for i := 0; i < len(args); i += 2 {
+		key := args[i]
+		value := args[i+1]
+		b.AddArg(key, value)
+	}
+	return b
+}
+
+// AddSend adds a send amount. (Only one send amount can be specified.)
+func (b *TxBuilder) AddSend(amount string) *TxBuilder {
+	if b == nil {
+		return nil
+	}
+	if amount == "" {
+		return b
+	}
+	b.send = amount
```

Comment:
this is not about "adding" a send, but "setting" it, meaning that we won't be able to "-send 1000ugnot -send 12345foo".
it sounds okay to me to limit "suggested amounts" to a single coin; but then we need to rename to SetSend().